### ISSUE TITLE
[FEATURE REQUEST] ES module build

### ISSUE THREAD

**Issue by hakimio** [newdash/newdash#942]
**Is your feature request related to a problem? Please describe.**
Right now `newdash` is published as [`commonJS` module](https://github.com/newdash/newdash/blob/1eb66c8d1bbd0fe9862f0df54f190fbfccac55c4/tsconfig.json#L5). This is perfectly fine when using it in `nodejs` server app but prevents tree-shaking when used on client side.
Following single function import will pull the whole `newdash` library into production bundle:
```ts
import {invert} from '@newdash/newdash';
```

**Describe the solution you'd like**
Either create a separate `esm` bundle package or use `esm` in the main package since `nodejs` added stable support for esm in v12 released on April 23, 2019 (5 years ago) and most other node libraries have already migrated to `esm`.

**Describe alternatives you've considered**
Use full path imports everywhere like `import invert from '@newdash/newdash/invert';`. This is not great alternative since you have to manually fix imports added by the IDE. Also, having all functions exported in root, enables discoverability since you get IDE autocompletion when searching for some function. IDE will show you all the functions statrting with "is" in the following case: `import {is_} from '@newdash/newdash';`.

**Additional context**
Angular framework throws warnings during production bundle build like the following:
```
[WARNING] Module '@newdash/newdash' used by 'src/app/shared/pipes/order-by.pipe.ts' is not ESM

CommonJS or AMD dependencies can cause optimization bailouts.
```

**Comment by Soontao** [comment]
yes, good point. let me try to convert `@newdash/newdash-es6` into esm module. then after node 18 out of maintenance we switch main package to esm.

**Comment by Soontao** [comment]
Hi @hakimio 

could you have a try with new released `@newdash/newdash-es6@5.23.1` ?

```js
import { first } from "@newdash/newdash-es6";
```

I want to wait node 18 end of life then migrate main distribution to esm.

**Comment by hakimio** [comment]
@Soontao 
While it's a lot better with the new `newdash-es6` version (39KB vs 200+KB and no more `CommonJS` warnings), there is still room for improvement.
- I am using five functions (startCase, invert, orderBy, kebabCase, isString) and I still see some unrelated imports (mostly from `.internal`) pulled into the bundle:

![image](https://github.com/newdash/newdash/assets/768105/6011ad17-6ac1-49f6-807b-f5f49b48dac0)

- A lot of functions are still not exported from the root `index.js`. For example, I can not do the following:
```ts
import { startCase, orderBy, kebabCase } from '@newdash/newdash-es6';
```
